release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -354,8 +366,23 @@ BufferedReader getPullRequestBody(String prNumber) throws Exception {
     BufferedReader runSubProcessAndGetOutputAsReader(String command) throws Exception {
       if (DEBUG != null) System.err.printf("Executed: %s%n", command);
       Process p = new ProcessBuilder("bash", "-c", command).start();
-      p.waitFor();
-      return new BufferedReader(new InputStreamReader(p.getInputStream(), StandardCharsets.UTF_8));
+
+      BufferedReader result = null;
+      try (BufferedReader br =
+          new BufferedReader(new InputStreamReader(p.getInputStream(), StandardCharsets.UTF_8)); ) {
+        String processOutput = br.lines().collect(Collectors.joining(System.lineSeparator()));
+        result = new BufferedReader(new StringReader(processOutput));
```

Comment:
Create another BufferedReader from the input stream via Process to keep the interface of this method.

Reply:
How about simply returning `processOutput` instead of `BufferedReader` to make things straightforward ?

Reply:
Agree. If we actually want to avoid deadlock due to a full of buffer with stream-based IO processing, we should spawn a thread and process the output in a separate thread.

But the size of output is expected to be small enough, we should read a whole output into a memory and return it directly, as you are doing.

Reply:
I agree with you. But in this PR. I leave the return value data type as the BufferedReader. The reasons I chose this are as follows:
- The `runSubProcessAndGetOutputAsReader` method has several callers. So, if I change the interface of this method, I need to change several places in this code.
- I want to focus on bugfix. The suggestion is refactoring. So I would like to do such a change in the other PR 

I would like to address this refactoring in a separate pull request.

What do you think?